react-effects/my-app/src/List.tsx

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { readItems, type Item } from './read';
+
+export function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState<Item[]>([]);
+  const [error, setError] = useState<unknown>();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+
+  useEffect(() => {
+    let ignore = false;
+    async function assignItems() {
+      try {
+        const i: Item[] = await readItems();
+        if (!ignore) {
+          setItems(i);
+          setIsLoading(false);
```

Comment:
This ends up calling `setIsLoading` twice because it's also called in the `finally`. That doesn't break anything, it's just wasteful and makes the code somewhat strange.